// hk-bus-eta/esm/nlb.js
import { isSafari } from "./utils";
export default function fetchEtas({ stopId, nlbId, }) {
    return fetch(`https://rt.data.gov.hk/v1/transport/nlb/stop.php?action=estimatedArrivals`, {
        body: JSON.stringify({
            routeId: nlbId,
            stopId,
            language: "zh",
        }),
        headers: {
            "Content-Type": "text/plain",
        },
        method: "POST",
        cache: isSafari ? "default" : "no-store",
    })
        .then((response) => response.json())
        .then(({ estimatedArrivals }) => {
        if (!estimatedArrivals)
            return [];
        return estimatedArrivals
            .filter((eta) => eta.estimatedArrivalTime)
            .map((e) => ({
            eta: e.estimatedArrivalTime.replace(" ", "T") + ".000+08:00",
            remark: {
                zh: "",
                en: "",
            },
            dest: {
                zh: "",
                en: "",
            },
            co: "nlb",
        }));
    });
}
